test/libsolidity/util/TestFileParser.h

```diff
@@ -111,9 +112,18 @@ class TestFileParser
 		char peek() const noexcept;
 
 		/// Returns true if the end of a line is reached, false otherwise.
-		bool isEndOfLine() const { return m_char == m_line.end(); }
+		bool isEndOfFile() const { return m_char == m_source.end(); }
 
-		std::string m_line;
+		/// Returns true if the next character is pointing to a new-line character, false otherwise.
+		/// Right now the scanner discards all newlines in TestFileParser::Scanner::readStream(..),
+		/// where Token::NewLine is NOT '\n', it is defined as `//`.
+		/// TODO: Fix this.
+		bool nextIsNewLine() const { return peek() == '/'; }
```

Comment:
Won't the check for `/` result in single slashes being treated as line endings?

If so then I think that creating a function for this is not a good idea. This is a hack so I'd rather have it used just in one place rather than risk someone using it and running into this problem.

I'd also at least add a slash in the middle of the line in one of the tests so that we can see how it actually behaves for side effects.

Reply:
Somehow I thought that defining these helper functions may increase the readability. But now I think that it is the best to remove this helpers and just put a very similar comment directly in `TestFileParser::Scanner::readLine()`.

Yes, the new-line handling is a mess and we should clean that up as soon as possible.  If there is a single `/` in the line, we will just get a parsing error - that means single or multiple `/` are just forbidden within side effects.

Reply:
However, I added a very simple test in `TestFileParserTest`.

Reply:
Interesting. I wasn't aware that `/` was forbidden inside the line. I wonder if it might be a problem down the line - what if an event emits something containing `/`?